### ISSUE TITLE
loop CREATE/DELETE on publisher endpoint doesn't close the subscriber connections

### ISSUE THREAD

**Issue by qinghon** [slact/nchan#691]
nchan no longer closes subscriber connections after creating/deleting channels multiple times in a loop

Reproducible configuration
Version 1.3.6 (debian12)
```
ii  libnginx-mod-nchan        1:1.3.6+dfsg-2                 amd64
ii  nginx                     1.22.1-9                       amd64
ii  nginx-common              1.22.1-9                       all  
```

nginx configuration
```
    nchan_shared_memory_size 2048M;

    server {
        listen       8443;
        server_name  localhost;

        location ~ /sub/(\w+)$ {
                nchan_subscriber;
                nchan_channel_id $1;
                nchan_websocket_ping_interval 60;
                nchan_message_buffer_length 0;
                nchan_channel_events_channel_id $1; #enables channel events for this location
        }
        location ~ /sub/server/callback/(\w+)$ {
                nchan_subscriber;
                nchan_channel_id "callback/$1";
                nchan_message_buffer_length 0;
        }

        location ~ /sub/server/(\w+)$ {
                return 403;
        }
        location ~ /pub/server/(\w+)$ {
                nchan_publisher;
                nchan_channel_id $1;
                nchan_message_buffer_length 0;
		nchan_channel_events_channel_id $1;
        }
        location ~ /pub/callback/(\w+)$ {
                nchan_publisher;
                nchan_channel_id "callback/$1";
                nchan_message_buffer_length 0;
                nchan_channel_events_channel_id $1; #enables channel events for this location
        }
        location ~ /pub/(\w+)$ {
                return 403;
        }
    }
```
Repeat operation
1. Start nginx
2. Use `ws://127.0.0.1:8443/sub/unknown_x86_64_xxxx` to connect to nginx
3. After about 1s, `curl -v -X DELETE -H "Accept: text/json" 127.0.0.1:8443/pub/server/unknown_x86_64_xxxx` should be observed. At this time, the websocket connection should be observed to be closed.
4. After repeating steps 2-3 about 10-20 times, the websocket link no longer closes automatically.

ps: Using redis has no impact on this result
note: websocket connect use page[https://wstool.js.org/](https://wstool.js.org/)

nginx log
```
127.0.0.1 - - [26/Jul/2024:12:12:05 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:05 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:09 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:09 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:12 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:12 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:15 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:15 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:17 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:17 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:19 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:19 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:21 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:21 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:23 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:23 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:24 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 101 30 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:25 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:27 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:27 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:29 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:29 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:30 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 101 30 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:32 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:32 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:34 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:34 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:35 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:35 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:37 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:37 +0000] "GET /sub/unknown_x86_64_xxxx HTTP/1.1" 410 12 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
127.0.0.1 - - [26/Jul/2024:12:12:40 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 76 "-" "curl/7.88.1" # From here on, subsequent DELETEs will no longer take effect and will always return 200
127.0.0.1 - - [26/Jul/2024:12:12:49 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:49 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:49 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:50 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:50 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
127.0.0.1 - - [26/Jul/2024:12:12:54 +0000] "DELETE /pub/server/unknown_x86_64_xxxx HTTP/1.1" 200 77 "-" "curl/7.88.1"
```

**Comment by slact** [comment]
I haven't been able to reproduce this. I suspect this is a websocket client issue that gives up after several retries.

**Comment by slact** [comment]
Thanks for the clarification, I'm able to reproduce this. Will have a fix shortly.

**Comment by slact** [comment]
Fixed in master, commit 1129b33ba58f309f580d7e046 . Will be released shortly.

**Comment by slact** [comment]
released in version 1.3.7
